src/Kinto.elm

```diff
@@ -211,18 +217,29 @@ errorDecoder =
         ("error" := Json.string)
 
 
-toKintoResponse : Task Http.RawError Http.Response -> Task Error Json.Value
-toKintoResponse response =
-    (response
-        |> Task.mapError NetworkError
-    )
-        `Task.andThen` handleResponse
+toKintoResponse : Http.Response -> Result Error Json.Value
+toKintoResponse ({ value, status, statusText } as response) =
+    (extractBody response)
+        `Result.andThen`
+            (\body ->
+                extractData body
+                    |> Result.formatError (extractError status statusText)
```

Comment:
Awesome! Thanks a lot, didn't know there was a function composition in elm (thought I should have guessed, as a functional language :P